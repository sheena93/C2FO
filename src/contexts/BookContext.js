/**
 * @component BookContextProvider
 * This component is a stateless component to create Context object
 * @return {element} returns BookContextProvider component
 */

import React, { createContext, useState } from 'react';
import uniqid from 'uniqid';

export const BookContext = createContext();

const BookContextProvider = ({ children }) => {
  const [books, setBooks] = useState([
    {
      id: 1,
      title: 'Number the Stars',
      author: 'Lois Lowry'
    },
    {
      id: 2,
      title: 'The Very Hungry Caterpillar',
      author: 'Eric Carle'
    }
  ]);

  const addBook = book => {
    setBooks([...books, { ...book, id: uniqid() }]);
  };

  const deleteBook = id => {
    setBooks(books.filter(book => book.id !== id));
  };

  return (
    <BookContext.Provider value={{ books, addBook, deleteBook }}>
      {children}
    </BookContext.Provider>
  );
};

export default BookContextProvider;