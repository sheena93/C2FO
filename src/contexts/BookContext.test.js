import React from 'react';
import { render, cleanup } from '@testing-library/react';
import BookContext from './BookContext'

afterEach(cleanup);
 
describe('BookContext', () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<BookContext />)
        expect(asFragment(<BookContext />)).toMatchSnapshot()
       })

       it('should equal to ', () => {
        const { getByTestId } = render(<BookContext  />); 
        expect(getByTestId('book')).toHaveTextContent(2)
       });
});
