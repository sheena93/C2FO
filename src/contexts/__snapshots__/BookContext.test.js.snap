// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should take a snapshot 1`] = `<DocumentFragment />`;

exports[`BookContext should take a snapshot 1`] = `<DocumentFragment />`;
